src/ray/raylet/node_manager.cc

```diff
@@ -419,10 +423,10 @@ void NodeManager::ProcessClientMessage(
     // Remove the dead worker from the pool and stop listening for messages.
     const std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
 
-    // This if statement distinguishes workers from drivers.
     if (worker) {
-      // Handle the case where the worker is killed while executing a task.
-      // Clean up the assigned task's resources, push an error to the driver.
+      // The client is a worker. Handle the case where the worker is killed
+      // while executing a task. Clean up the assigned task's resources, push
+      // an error to the driver.
```

Comment:
I can address this in a different PR, but I noticed that when the client dies, we should be removing the in-progress task from the queues, which we are not doing.